WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/download/details/BackupDownloadDetailsViewModel.kt

```diff
@@ -49,38 +48,40 @@ class BackupDownloadDetailsViewModel @Inject constructor(
-            _uiState.value = buildContentUiState(availableItems)
-        }
-    }
-
-    private suspend fun buildContentUiState(items: List<BackupAvailableItem>): Content {
-        return withContext(bgDispatcher) {
-            val availableItemsListItems: List<ListItemUiState> = items.map {
-                ListItemUiState(
-                        availableItemType = it.availableItemType,
-                        label = UiStringRes(it.labelResId),
-                        checked = true,
-                        onClick = { onItemClicked(it.availableItemType) }
+            val availableItems = availableItemsProvider.getAvailableItems()
+            // todo: annmarie move this to a useCase
+            val activityLogModel = activityLogStore.getActivityLogItemByActivityId(activityId)
+            if (activityLogModel != null) {
+                _uiState.value = Content(
+                        items = stateListItemBuilder.buildDetailsListStateItems(
+                                availableItems = availableItems,
+                                activityLogModel = activityLogModel,
+                                onCreateDownloadClick = this@BackupDownloadDetailsViewModel::onCreateDownloadClick,
+                                onCheckboxItemClicked = this@BackupDownloadDetailsViewModel::onCheckboxItemClicked
+                        )
                 )
+            } else {
+                // todo: annmarie - snackbar message here and leave the wizard?
             }
-            // todo: annmarie - swap out the placeholder for date from record
-            Content(
-                    description = UiStringRes(R.string.backup_download_details_description),
-                    items = availableItemsListItems
-            )
         }
     }
 
-    private fun onItemClicked(backupAvailableItemType: BackupAvailableItemType) {
-        // todo: annmarie update the checkboxes - keep a running list of selected checkboxes, so
-        // they can be persisted on rotation
+    private fun onCreateDownloadClick() {
+        // todo: annmarie implement onActionButtonClicked
+    }
+
+    private fun onCheckboxItemClicked(itemType: JetpackAvailableItemType) {
         (_uiState.value as? Content)?.let { content ->
-            val updatedList = content.items.map { itemUiState ->
-                if (itemUiState.availableItemType == backupAvailableItemType) {
-                    itemUiState.copy(checked = !itemUiState.checked)
+            val updatedList = content.items.map { contentState ->
+                if (contentState.type == CHECKBOX) {
+                    contentState as CheckboxState
+                    if (contentState.availableItemType == itemType) {
+                        contentState.copy(checked = !contentState.checked)
+                    } else {
+                        contentState
+                    }
                 } else {
-                    itemUiState
+                    contentState
```

Comment:
Great suggestion, I'll rewrite in the next PR because filter can be tricky. The above example snippet will filter out all values and I'll be left with a single checkbox on the screen , but I see where you were going. Thanks 👍